File: gpao-tex-ui/source/main/src/app/services/product/models/pilotage-efficience.ts
/* tslint:disable */
/* eslint-disable */
import { Produit } from '../models/produit';
import { Site } from '../models/site';
export interface PilotageEfficience {
  archived?: boolean;
  creationDate?: string;
  creatorIdentifier?: string;
  deletionDate?: string;
  efficienceJour1?: number;
  efficienceJour2?: number;
  efficienceJour3?: number;
  efficienceJour4?: number;
  efficienceJour5?: number;
  efficienceJour6?: number;
  efficienceJour7?: number;
  id?: number;
  latestUpdaterId?: string;
  modificationDate?: string;
  produit?: Produit;
  site?: Site;
  version?: number;
}

File: gpao-tex-core/produit/src/main/java/com/gpro/consulting/produit/model/PilotageEfficience.java
package com.gpro.consulting.produit.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "eb_pilotage_efficience")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PilotageEfficience extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "site_id")
    private Site site;

    @ManyToOne
    @JoinColumn(name = "eb_produit_id")
    private Produit produit;

    @Id
    @GeneratedValue(generator = "PILOTAGE_EFFICIENCE_ID_GENERATOR", strategy = GenerationType.SEQUENCE)
    @SequenceGenerator(name = "PILOTAGE_EFFICIENCE_SEQ", sequenceName = "PILOTAGE_EFFICIENCE_SEQ", allocationSize = 1)
    private Long id;
    private Double efficienceJour1;
    private Double efficienceJour2;
    private Double efficienceJour3;
    private Double efficienceJour4;
    private Double efficienceJour5;
    private Double efficienceJour6;
    private Double efficienceJour7;
}
